java/client/src/main/java/glide/api/RedisClusterClient.java

```diff
@@ -53,17 +55,19 @@ public CompletableFuture<ClusterValue<Object>> customCommand(String[] args) {
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public CompletableFuture<ClusterValue<Object>> customCommand(String[] args, Route route) {
         return commandManager.submitNewCommand(
-                CustomCommand,
-                args,
-                route,
-                response ->
-                        route.isSingleNodeRoute()
-                                ? ClusterValue.ofSingleValue(handleObjectOrNullResponse(response))
-                                : ClusterValue.ofMultiValue(
-                                        (Map<String, Object>) handleObjectOrNullResponse(response)));
+                CustomCommand, args, route, response -> handleCustomCommandResponse(route, response));
+    }
+
+    protected ClusterValue<Object> handleCustomCommandResponse(Route route, Response response) {
+        if (route.isSingleNodeRoute()) {
+            return ClusterValue.ofSingleValue(handleObjectOrNullResponse(response));
+        }
+        if (response.hasConstantResponse()) {
```

Comment:
We should have a test for this in java/client/src/test/java/glide/api/RedisClusterClientTest.java

Reply:
Tested in `custom_command_returns_single_value_on_constant_response` (a part of another PR), already merged on upstream